Modules/Scripted/VPAWModel/VPAWModel.py

```diff
@@ -361,74 +381,395 @@ def onApplyButton(self):
+                slicer.util.errorDisplay(mesg, "Install Error")
+                print(mesg)
+                return False
+            try:
+                # Install missing packages
+                with BusyCursor():
+                    slicer.util.pip_install(["--upgrade", "pip", "setuptools", "wheel"])
+                    slicer.util.pip_install(
+                        [pip_install_name for _, pip_install_name in needs_installation]
+                    )
+                    installed_modules = [
+                        (
+                            module_name,
+                            pip_install_name,
+                            importlib.import_module(module_name),
+                        )
+                        for module_name, pip_install_name in needs_installation
+                    ]
+            except ModuleNotFoundError as e2:
+                slicer.util.errorDisplay(
+                    "\n".join(
+                        [
+                            f"Unable to install package{plural}.",
+                            "Check the console for details.",
+                        ]
+                    ),
+                    "Install Error",
+                )
+                print(e2)
+                return False
+
+        # All dependencies were successfully imported
+        self.showInstalledModules(installed_modules)
+        return True
+
+    def runPediatricAirwayAtlas(
+        self,
+        pediatricAirwayAtlasDirectory,
+        vPAWRootDirectory,
+        vPAWModelsDirectory,
+        patientPrefix,
     ):
         """
-        Run the processing algorithm.
+        Run the Pediatric Airway Atlas pipeline.
         Can be used without GUI widget.
-        :param inputVolume: volume to be thresholded
-        :param outputVolume: thresholding result
-        :param imageThreshold: values above/below this threshold will be set to 0
-        :param invert: if True then values above the threshold will be set to 0,
-               otherwise values below are set to 0
-        :param showResult: show output volume in slice viewers
-        """
-
-        if not inputVolume or not outputVolume:
-            raise ValueError("Input or output volume is invalid")
 
-        import time
+        Parameters
+        ----------
+        pediatricAirwayAtlasDirectory : str
+            This is the directory containing the source code for pediatric_airway_atlas.
+        vPAWRootDirectory : str
+            This is the input/output directory, e.g., "path/to/vpaw-data-root".  It must
+            contain FilteredControlBlindingLogUniqueScanFiltered.xls, images/*, and
+            landmarks/*.
+        vPAWModelsDirectory : str
+            It must contain a file with name like "116(158.10-38.AM.24.Mar).pth".
+        patientPrefix : str
+            Process only files with this prefix.  Blank means all files.
+        """
+
+        # If self.pediatric_airway_atlas is not yet set, then see if we can set it.
```

Comment:
I like that it now tries to link if I forget to press "link" before pressing "run"

Reply:
Yes, with the `pediatric_airway_atlas` value automatically repopulating at application start, I wanted a mechanism whereby the `sys.path` linking to it was also automated.  But in the case where it was populated but dependencies hadn't yet been installed -- if the user had typed in a value but hadn't hit the Link button -- I didn't want to force the user through the dependency installation at application restart.  Waiting until the user hits a button (`Run` or `Link`) was a nice way to achieve these priorities.